Add markers for cluster APIs. (#9110)
opensearch-project/documentation-website: _api-reference/cluster-api/cluster-voting-configuration.md

---
layout: default
title: Cluster voting configuration
nav_order: 55
parent: Cluster APIs
---

> # Cluster voting configuration

Reviewer: ```suggestion
# Cluster voting configuration exclusions
```

**Introduced 1.0**
{: .label .label-purple }

> The Cluster voting configuration API updates and clears voting configurations.

Reviewer: ```suggestion
The Cluster Voting Configuration Exclusions API adds or removes cluster-manager-eligible nodes from the list of excluded nodes in the voting configuration.
```

# Endpoints

> The `POST` method updates the cluster voting configuration by excluding certain node IDs or names.

Reviewer: ```suggestion
The `POST` method excludes the specified nodes from the cluster voting configuration:
```

<!-- spec_insert_start
api: cluster.post_voting_config_exclusions
component: endpoints
omit_header: true
-->
```json
POST /_cluster/voting_config_exclusions
```
<!-- spec_insert_end -->

> The `DELETE` method clears any cluster voting configuration exclusions.

Reviewer: ```suggestion
The `DELETE` method removes all exclusions from the voting configuration:
```

<!-- spec_insert_start
api: cluster.delete_voting_config_exclusions
component: endpoints
omit_header: true
-->
```json
DELETE /_cluster/voting_config_exclusions
```
<!-- spec_insert_end -->

<!-- spec_insert_start
api: cluster.post_voting_config_exclusions
component: query_parameters
-->
## Query parameters

> The following table lists the available query parameters. All query parameters are optional.

Reviewer: `node_ids` or `node_names` is required for the POST method.  Please update the spec.

Author: We can't update the spec to require both of them, since this would change the behavior of the API. I'll update the descriptions to say that one or the other is required.

| Parameter | Data type | Description |
| :--- | :--- | :--- |
| `node_ids` | List or String | A comma-separated list of node IDs to exclude from the voting configuration. When using this setting, you cannot also specify `node_names`. |
| `node_names` | List or String | A comma-separated list of node names to exclude from the voting configuration. When using this setting, you cannot also specify `node_ids`. |

> | `timeout` | String | When adding a voting configuration exclusion, the API waits for the specified nodes to be excluded from the voting configuration before returning a response. If the timeout expires before the appropriate condition is satisfied, the request fails and returns an error. |

Reviewer: Default is 30 seconds. Please update the spec.

Author: @kolchfa-aws: How are you getting these defaults? Is there somewhere else in the code besides the spec where defaults are set?

Author: I've already set defaults where applicable.

> 

Author: It does not.

> 

Reviewer: I see it here https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/action/admin/cluster/configuration/ClearVotingConfigExclusionsRequest.java#L59
